File: new_Anal_frame/new_analysis_code/histDumper.cc
// vim: set sts=4 sw=4 fdm=marker et:
#include <iostream>
#include <array>
#include "math.h"
#include "cstdio.h"
#include "gROOT.h"
#include "TChain.h"
#include "TFile.h"
#include "TTree.h"
#include "TString.h"
#include "CompartmentObject.h"
#include "waferGeom.h"
#include "TH1D.h"
#include "TProfile.h"

const int G_nLayers = 8;
const int G_inputEnergy = 125;


//int nHits;

void regTree()
{//{{{
    // Copy the settings from makePlots
    std::string treeName = "HGCLayer";
    TChain *ch = new TChain(treeName);
    ch -> Add("./HGCLayerTree.root");
    
}//}}}

void histDumper(){
    // Follow the main function in histDumper.py
    regTree();
    TFile *fout = TFile("dumpedHist.root",'RECREATE');
    // Import simple plots to be drawn
    
    TFile *handle = open("./histDef.py");
    exec(handle);
    handle -> Close();


    
    //For complicate plots 
    TH1D *h1_erawHit = new TH1D();
    std::array<int,>ringRadii = ('f',{10,19,29,34,38,51,58,68,77});
    TProfile *pr_twoPtCor_rings = new ("pr_twoPtCor_rings","",9,ringRadii);
    
    Long64_t nentries = ch -> GetEntries();
    for (int evt = 0; ){
        for (){
            for (){
            }
        }
    }

    fout -> Write();
    fout -> Close();
}

int main(){
    histDumper();
    return 0;
}

File: new_Analysis_framework/new_analysis_code/histDumper.py
#!/usr/bin/env python

import math

from ROOT import gROOT, TChain, TTree, TFile
from ROOT import TH1D, TProfile
from array import array

gROOT.LoadMacro("./CompartmentObject.h")
gROOT.LoadMacro("./waferGeom.h")
from ROOT import CompartmentObject, HexGeometry

G_nLayers       = 8
G_inputEnergy   = 125 

def main(treeName="HGCLayer"):

    # Chain input files.
    ch = TChain(treeName)
    ch.Add("./HGCLayerTree.root")

    # 
    fout = TFile("dumpedHist.root",'RECREATE')

    # Import simple plots to be drawn
    plots = []
    handle = open("./histDef.py")
    exec(handle)
    handle.close()

    for p in plots:
        for step in p[1:-1]:
            ch.Draw(step[0],step[1],"goff")
            pass

    # For complicate plots.
    h1_erawHit = TH1D("h1_erawHit","",8,1,9)
    ringRadii  = array('f',[10,19,29,34,38,50,51,58,68,77])
    pr_twoPtCor_rings = TProfile("pr_twoPtCor_rings","",9,ringRadii)

    for ev in ch:
        for iL in range(G_nLayers):
            layerBranch = getattr(ev,"layer{0}".format(iL))
            for iHit in range(layerBranch.GetNhits()):
                h1_erawHit.Fill(iL+1,layerBranch.GetErawHit(iHit))
                for jHit in range(iHit+1,layerBranch.GetNhits()):
                    dist = math.sqrt((layerBranch.GetXHit(iHit)-layerBranch.GetXHit(jHit))**2
                                    +(layerBranch.GetYHit(iHit)-layerBranch.GetYHit(jHit))**2)
                    pr_twoPtCor_rings.Fill(dist,layerBranch.GetErawHit(iHit)*layerBranch.GetErawHit(jHit))

    fout.Write()
    fout.Close()

if __name__ == '__main__':
    main()
